docs/workflow/debugging/coreclr/debugging-crossgen2.md

```diff
@@ -0,0 +1,65 @@
+How to Debug Crossgen2
+=================
+
+Crossgen2 brings with it a number of new challenges for debugging the compilation process. Fortunately, in addition to challenges, crossgen2 is designed to enhance various parts of the debugging experience.
+
+Important concerns to be aware of when debugging Crossgen2
+---------------------------------
+
+- Other than the JIT, Crossgen2 is a managed application
+- By default Crossgen2 uses a multi-core compilation strategy
+- A Crossgen2 process will have 2 copies of the JIT in the process at the same time, the one used to compile the target, and the one used to compile Crossgen2 itself.
+- Crossgen2 does not parse environment variables for controlling the JIT (or any other behavior), all behavior is controlled via the command line
+- The Crossgen2 command line as generated by the project system is quite complex
+
+Built in debugging aids in Crossgen2
+---------------------------------
+
+- When debugging a multi-threaded component of Crossgen2 and not investigating a multi-threading issue itself, it is generally advisable to disable the use of multiple threads.
+To do this use the `--parallelism 1` switch to specify that the maximum parallelism of the process shall be 1.
+
+- When debugging the behaviour of compiling a single method, the compiler may be instructed to only compile a single method. This is done via the various --singlemethod options
```

Comment:
Nit - exploring British spelling (behaviour) :D?

Reply:
:) More like I recognise both forms and so I tend to gloss over such details. Also, my spell checker didn't notice.  :)